src/main/java/stirling/software/SPDF/repository/UserRepository.java

```diff
@@ -3,10 +3,19 @@
 import java.util.Optional;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import stirling.software.SPDF.model.User;
 
 public interface UserRepository extends JpaRepository<User, String> {
+    @Query(
+            "SELECT u FROM User u WHERE "
+                    + "(:ignoreCase = false AND u.username = :username) OR "
+                    + "(:ignoreCase = true AND LOWER(u.username) = LOWER(:username))")
```

Comment:
Rather than awkward settings here you can just use JPAs auto 

Optional<User> findByUsernameIgnoreCase(String username);


And have the toggle on the service side 
What do you think?